Mage/src/main/java/mage/abilities/effects/common/ExileThenReturnTargetEffect.java

```diff
@@ -23,23 +25,34 @@ public class ExileThenReturnTargetEffect extends OneShotEffect {
     private final boolean yourControl;
     private final boolean textThatCard;
     private final PutCards putCards;
+    private final Effect afterEffect;
 
     public ExileThenReturnTargetEffect(boolean yourControl, boolean textThatCard) {
         this(yourControl, textThatCard, PutCards.BATTLEFIELD);
     }
 
     public ExileThenReturnTargetEffect(boolean yourControl, boolean textThatCard, PutCards putCards) {
+        this(yourControl, textThatCard, putCards, null);
+    }
+
+    public ExileThenReturnTargetEffect(boolean yourControl, boolean textThatCard, Effect afterEffect) {
+        this(yourControl, textThatCard, PutCards.BATTLEFIELD, afterEffect);
+    }
+
+    public ExileThenReturnTargetEffect(boolean yourControl, boolean textThatCard, PutCards putCards, Effect afterEffect) {
```

Comment:
For infrequent usages, I prefer using a chaining method `withAfterEffect()` rather than cluttering the constructors.

I also wonder if it's possible to pass in a condition and an effect, so that a custom effect isn't necessary